src/runtime/interfaceobject.cs

```diff
@@ -71,7 +71,21 @@ public static IntPtr tp_new(IntPtr tp, IntPtr args, IntPtr kw)
                 return IntPtr.Zero;
             }
 
-            return CLRObject.GetInstHandle(obj, self.pyHandle);
+            return self.WrapObject(obj);
+        }
+
+        /// <summary>
+        /// Wrap the given object in an interface object, so that only methods
+        /// of the interface are available. The wrapped object is available
+        /// through properties in both converted/encoded (__implementation__)
+        /// and raw form (__raw_implementation__).
+        /// </summary>
+        public IntPtr WrapObject(object impl)
+        {
+            var objPtr = CLRObject.GetInstHandle(impl, pyHandle);
+            Runtime.PyObject_SetAttrString(objPtr, "__implementation__", Converter.ToPython(impl));
+            Runtime.PyObject_SetAttrString(objPtr, "__raw_implementation__", CLRObject.GetInstHandle(impl));
```

Comment:
These two must be computed only on demand.

Reply:
OK. I fixed it by implementing `tp_getattro`. Note that the properties show up when you do `dir` on an interface object. If that is desired I should probably go the long route via `tp_getset`.